### PR TITLE
Rename timeout to timeout_idle for $worker_config_save

### PR DESCRIPTION
**Pull request by M-Kusumgar**
There were a lot of timeouts in the code. With this PR I hope to change some of those timeouts (ones relevant to $worker_config_save) to timeout_idle. Hopefully I have got all the relevant places in this PR.

### CODE REVIEW

File: R/rrq_controller.R [hunk 1]
```diff
@@ -999,11 +999,11 @@ rrq_controller <- R6::R6Class(
     ##'   wait for the background process to respond to SIGTERM before
     ##'   we stop the worker. Only used for tasks queued with
     ##'   `separate_process` `TRUE`.
-    worker_config_save = function(name, time_poll = NULL, timeout = NULL,
+    worker_config_save = function(name, time_poll = NULL, timeout_idle = NULL,
                                   queue = NULL, heartbeat_period = NULL,
                                   verbose = NULL, overwrite = TRUE,
                                   timeout_poll = 1, timeout_die = 2) {
-      worker_config_save(self$con, private$keys, name, time_poll, timeout,
+    worker_config_save(self$con, private$keys, name, time_poll, timeout_idle,
```
Comment (richfitz) [inline R/rrq_controller.R:1006]:
```suggestion
      worker_config_save(self$con, private$keys, name, time_poll, timeout_idle,
```

Something happened to the indenting here...

File: R/rrq_controller.R [hunk 2]
```diff
@@ -963,7 +963,7 @@ rrq_controller <- R6::R6Class(
     ##'   killed with an interrupt.  The default should be good for most
     ##'   uses, but shorter values are used for debugging.
     ##'
-    ##' @param timeout Optional timeout to set for the worker.  This is
+    ##' @param timeout_idle Optional timeout to set for the worker.  This is
     ##'   (roughly) equivalent to issuing a \code{TIMEOUT_SET} message
     ##'   after initialising the worker, except that it's guaranteed to be
     ##'   run by all workers.
```
Comment (r-ash) [inline R/rrq_controller.R:969]: I wonder if worth adding here some more details about what will happen when the timeout is hit. From the `TIMEOUT_SET` docs `Sets the length of time after which the worker will exit if it has not processed a task`

**Review by richfitz — CHANGES_REQUESTED** [review]
Thanks, that name is much clearer. Can you bump the version number up in the `DESCRIPTION` file please?

**Review by r-ash — APPROVED** [review]
This looks good to me, I think worth fixing that indentation indentation. And a small comment about adding some additional info to the docs.

**Review by richfitz — APPROVED** [review]

**Pull request merged**